src/@context/UserPreferences.tsx

```diff
@@ -40,97 +45,161 @@ interface UserPreferencesValue {
-  return (
-    isBrowser &&
-    window.localStorage.setItem(localStorageKey, JSON.stringify(values))
-  )
-}
-
 function UserPreferencesProvider({
   children
 }: {
   children: ReactNode
 }): ReactElement {
   const { appConfig } = useMarketMetadata()
-  const localStorage = getLocalStorage()
   // Set default values from localStorage
-  const [debug, setDebug] = useState<boolean>(localStorage?.debug || false)
+  const [debug, setDebug] = useState<boolean>(
+    getCookieValue('debug') === 'true'
+  )
   const [currency, setCurrency] = useState<string>(
     localStorage?.currency || 'EUR'
   )
   const [locale, setLocale] = useState<string>()
-  const [bookmarks, setBookmarks] = useState(localStorage?.bookmarks || [])
+  const [bookmarks, setBookmarks] = useState(
+    getCookieValue('bookmarks')?.split(',') || []
+  )
   const [chainIds, setChainIds] = useState(
-    localStorage?.chainIds || appConfig.chainIds
+    getCookieValue('chainIds')?.split(',').map(Number) || appConfig.chainIds
   )
   const { defaultPrivacyPolicySlug, showOnboardingModuleByDefault } = appConfig
 
   const [privacyPolicySlug, setPrivacyPolicySlug] = useState<string>(
-    localStorage?.privacyPolicySlug || defaultPrivacyPolicySlug
+    getCookieValue('privacyPolicySlug') || defaultPrivacyPolicySlug
   )
 
   const [showPPC, setShowPPC] = useState<boolean>(
-    localStorage?.showPPC !== false
+    getCookieValue('showPPC') !== 'false'
   )
 
   const [allowExternalContent, setAllowExternalContent] = useState<boolean>(
-    localStorage?.allowExternalContent || false
+    getCookieValue('allowExternalContent') === 'true'
   )
 
   const [automationWallet, setAutomationWallet] = useState<string>(
-    localStorage?.automationWalletJSON || ''
+    typeof getCookieValue('automationWallet') !== 'undefined'
+      ? JSON.parse(getCookieValue('automationWallet'))?.automationWalletJSON
+      : ''
   )
 
   const [automationWalletMode, setAutomationWalletMode] =
     useState<AUTOMATION_MODES>(
-      localStorage?.automationWalletMode || AUTOMATION_MODES.SIMPLE
+      typeof getCookieValue('automationWallet') !== 'undefined'
+        ? JSON.parse(getCookieValue('automationWallet'))?.automationWalletMode
+        : AUTOMATION_MODES.SIMPLE
     )
 
   const [showOnboardingModule, setShowOnboardingModule] = useState<boolean>(
-    localStorage?.showOnboardingModule ?? showOnboardingModuleByDefault
+    (typeof getCookieValue('onboardingModule') !== 'undefined' &&
+      JSON.parse(getCookieValue('onboardingModule'))?.showOnboardingModule ===
+        'true') ??
+      showOnboardingModuleByDefault
   )
 
   const [onboardingStep, setOnboardingStep] = useState<number>(
-    localStorage?.onboardingStep || 0
+    typeof getCookieValue('onboardingModule') !== 'undefined'
+      ? ~~JSON.parse(getCookieValue('onboardingModule'))?.onboardingStep
+      : 0
   )
+  const twoMonthsFromNow = new Date()
+  twoMonthsFromNow.setMonth(twoMonthsFromNow.getMonth() + 2)
```

Comment:
suggestion: move this to `@utils/cookies`, similar to the 365 days. Could be some defaults to import when setting cookie length